src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -29,18 +29,91 @@
  */
 package com.rultor.agents;
 
+import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Ordering;
 import com.rultor.spi.SuperAgent;
+import com.rultor.spi.Talk;
 import com.rultor.spi.Talks;
 import java.io.IOException;
+import java.util.UUID;
+import org.xembly.Directives;
 
 /**
  * Adds index to all the requests received.
+ *
  * @author Krzysztof Krason (Krzysztof.Krason@gmail.com)
  * @version $Id$
  */
+@SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
 public final class IndexesRequests implements SuperAgent {
     @Override
     public void execute(final Talks talks) throws IOException {
-        throw new UnsupportedOperationException();
+        if (talks != null) {
+            int idx = this.index(talks);
+            for (final Talk talk : talks.active()) {
+                final int current = idx + 1;
+                final String index = "index";
+                if (talk.read().nodes("/talk/request").isEmpty()) {
+                    talk.modify(new Directives().xpath("/talk").addIf("request")
```

Comment:
this is wrong. we DON'T add requests when they are absent. see our discussion in this pull request: https://github.com/yegor256/rultor/pull/696